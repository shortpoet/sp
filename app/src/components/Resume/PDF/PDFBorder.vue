<template>
  <hr 
    :style="styleObject"
  />
  <!-- <hr /> -->
</template>
<script>
export default {
  name: 'PDFBorder',
  props: {
    size: {
      type: Number,
      default: 1
    },
    marginX: {
      type: Number,
      default: 2
    },
    color: {
      type: String,
      default: null
    }
  },
  computed: {
    styleObject () {
      let s = this.size
      let m = this.marginX
      if (this.color) {
        return {
          borderWidth: `${.25*(s)}rem`,
          margin: `${.25*(m)}rem`,
          color: this.color
        }
      } else {
        return {
          borderWidth: `${.25*(s)}rem`,
          margin: `${.25*(m)}rem`,
        }
      }
    }
  }
}
</script>
<style scoped lang="scss">
  hr {
    border: .25rem solid;
    border-radius: .25rem;
    color: $primary;
    margin: 0rem 2rem 0rem 2rem;
  }

  // // this apparently makes a box
  // hr {
  //   height: 1rem;
  //   width: 1rem;
  //   border: .25rem solid;
  //   border-radius: .25rem;
  //   color: $primary;
  //   margin: 0rem 2rem 0rem 2rem;
  // }
</style>