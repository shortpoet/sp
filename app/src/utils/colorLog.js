const chalk = require('chalk')
import {inspect} from 'util'

// const _log  = () => Function.prototype.bind.call(console.log, console, 'LOG_ALIAS')

export const log = (color, message) => console.log(chalk[`${color}`](`${inspect(message)}`))

export const colorLog = (message, color, background) => {
  color = color || "black";
  if (background) {
    console.log(`%c` + `${message}`, `color:` + `${color};background:${background}`)
  } else {
    console.log(`%c` + `${message}`, `color:` + `${color}`)
  }
}

// module.exports =  {
//   log,
//   colorLog
// }
