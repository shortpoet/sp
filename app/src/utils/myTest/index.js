import createWrapper from './test.utils.createWrapper'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'
import propFinder from './test.utils.propFinder'
import propsFinder from './test.utils.propsFinder'
import propMocker from './test.utils.propMocker'
import propsMocker from './test.utils.propsMocker'
import textMatcher from './test.utils.textMatcher'
import textMatcherFactory from './test.utils.textMatcherFactory'

export default {
  createWrapper,
  factory,
  factoryShallow,
  propFinder,
  propsFinder,
  propMocker,
  propsMocker,
  textMatcher,
  textMatcherFactory
}
