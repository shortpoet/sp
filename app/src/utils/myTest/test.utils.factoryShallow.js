import { shallowMount } from '@vue/test-utils'

export const factoryShallow = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    return shallowMount(component, options)
  }

module.exports = factoryShallow