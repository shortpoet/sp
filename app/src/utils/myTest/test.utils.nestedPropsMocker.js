

// // this is incomplete
// // maybe try and develop using TDD  
  
// export const nestedPropsMocker = (props) => {

//   console.log(props)

//   let propsData = {}


//   const handleArray = (props) => {
//     propsData = {}
//     props.map(prop => {
//       propsData[`${prop}`] = `sample ${prop}`
//     })
//     return {
//       propsData
//     }
//   }


//   const handleNested = (props, propsData, previousKey, nextKey) => {
//     console.log(props, propsData)

//     let firstLevel
//     if (previousKey === null && nextKey === null) {
//       firstLevel = true
//     }

//     // spread to clone so the splice works on subsequent tests
//     let values = [...props]

//     const length = parseInt(values.splice(-1, 1)[0])
//     console.log(length)
//     const iterator = [...Array(length).keys()]

//     if (firstLevel === true) {
//       previousKey = values.splice(0, 1)[0]
//       propsData[`${previousKey}`] = []
//     } else {
//       console.log('#### not first run ####')
//       console.log(propsData)
//       console.log(previousKey)
//       nextKey = values.splice(0, 1)[0]
//       console.log(nextKey)
//       propsData[`${previousKey}`][`${nextKey}`] = []
//     }


//     let keyDict

//     for (let i of iterator) {
//       keyDict = {}
//       values.map(prop => {
//         if (Array.isArray(prop)) {
//           console.log(prop)
//           console.log('#### calling handleNested from nested ####')
//           // this would only work for one level nested
//           // TODO maybe refactor for infinite recursion
//           console.log(previousKey)
//           console.log(nextKey)
//           console.log(propsData)
//           handleNested(prop, propsData, previousKey, nextKey)
//         } else {
//           console.log(prop)
//           keyDict[`${prop}`] = `sample ${prop}`
//         }
//       })
//       if (firstLevel === true) {
//         propsData[`${previousKey}`].push(keyDict)
//         firstRun = false
//       } else {
//         console.log(propsData)
//         console.log(previousKey)
//         console.log(nextKey)
//         console.log(propsData.experiences)
//         console.log(propsData.experiences.jobs)
//         propsData[`${previousKey}`][`${nextKey}`].push(keyDict)
//       }
//     }
//     return {
//       propsData
//     }
//   }

//   const isNested = props.some(p => Array.isArray(p))
//   const previousKey = null
//   const nextKey = null

//   if (isNested) {
//     console.log('#### is nested ####')
//     handleNested(props, propsData, previousKey, nextKey)
//   } else  {
//     handleArray(props)
//   }
// }
