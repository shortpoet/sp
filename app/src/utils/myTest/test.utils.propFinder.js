import {hardResume} from '@/assets/resume.js'

export const propFinder = (prop) => {
  return {
    propsData: {
      [`${prop}`]: hardResume[`${prop}`]
    }
  }
}

module.exports = propFinder