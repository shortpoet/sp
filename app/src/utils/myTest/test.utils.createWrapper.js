import { createLocalVue } from '@vue/test-utils'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'

// import VueRouter from 'vue-router'
import Vuex from 'vuex'
import PortalVue from 'portal-vue'

import createStore from '@/store/createStore'
// import createMockRouterConfig from '@/router/createMockRouterConfig'
// import paths from '@/router/paths.js'

import { log } from '@/utils/colorLog'
export const createWrapper  = (
  component,
  /* istanbul ignore next */
  options = {},
  // but why don't i have to ignore the empty  or even unprovided resumeStoreOtions
  // might need to figure this out but setting params to ignore for now
  resumeStoreOptions = {},
  // routerOptions = {},
  isShallow = false,
  ) => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(PortalVue)
    const mockStoreResume = createStore.createMocks().createStoreResumeMocks(resumeStoreOptions)
    const storeConfig = createStore.createMocks({modules: {resume: mockStoreResume}})

    // Using router is untested makes tests fail
    // and just doesn't work for now
    // localVue.use(VueRouter)
    // const routerConfig = createMockRouterConfig.createMockRouterConfig(routerOptions)
    // const router = new VueRouter(routerConfig)

    // log('green', mockStoreResume)
    // log('yellow', storeConfig.modules.resume)
    const store = new Vuex.Store(storeConfig)
    // log('red', store)
    // log('blue', store._mutations['resume/SET_RESUME_RAW'][0]())
    // console.log(options)
    return isShallow ? 
    factoryShallow(component, {
      localVue,
      store,
      // router,
      ...options
    }) : 
    factory(component, {
      localVue,
      store,
      // router,
      ...options
    })
}

// export default createWrapper
// when this was set to export default a console log(createWrapper) that was in
// test.utils.textMatcher line 19 was being executed from every module that called createWrapper
module.exports = createWrapper