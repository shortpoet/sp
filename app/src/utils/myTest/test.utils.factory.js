import { mount } from '@vue/test-utils'
import { log } from '@/utils/colorLog'
export const factory = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    // this only breaks when testing
    // probably should comment out of real func (which this is (now) hehe)
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    // console.log(options)
    // console.log(options.mocks)
    // console.log(options.store)
    // console.log(options.localVue)
    // log('blue', options.stubs)
    // log('green', options.propsData)
    // console.log(mount)
    return mount(component, options)
  }

module.exports = factory

// export default factory