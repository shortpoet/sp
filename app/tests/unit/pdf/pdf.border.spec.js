import * as myTest from '@/utils/myTest'
const { createWrapper } = myTest.default
import PDFBorder from '@/components/Resume/PDF/PDFBorder'
import { cloneDeep } from 'lodash'

describe('PDFBorder.vue', () => {

  const component = PDFBorder
  const mountOptions = {
    propsData: {
      size: 1,
      marginX: 2,
      color: null
    }
  }
  let wrapper

  let color
  let styleObject 
  let expected

  beforeEach(() => {
    wrapper = createWrapper(component, mountOptions)
    color = null
    styleObject = {
      borderWidth: ``,
      margin: ``,
    }
    expected = `border-width: 0.25rem; margin: 0.5rem;`
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('renders hr with correctly calculated style with color', () => {

    const _mountOptions = cloneDeep(mountOptions)
    _mountOptions.propsData.color = 'green'
    expected += ` color: green`

    wrapper = createWrapper(component, _mountOptions)

    expect(wrapper.find('hr').attributes().style).toMatch(expected)

  })
  it('renders hr with correctly calculated style without color', () => {
    
    expect(wrapper.find('hr').attributes().style).toMatch(expected)

  })
  it('matches snapshot', () => {

    expect(wrapper.html()).toMatchSnapshot()

  })
})
