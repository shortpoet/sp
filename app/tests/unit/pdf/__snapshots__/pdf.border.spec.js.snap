// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PDFBorder.vue matches snapshot 1`] = `<hr style="border-width: 0.25rem; margin: 0.5rem;">`;
